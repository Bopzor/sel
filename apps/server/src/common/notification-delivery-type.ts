export enum NotificationDeliveryType {
  email = 'email',
  push = 'push',
}
