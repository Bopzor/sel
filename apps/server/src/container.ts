import { createContainer } from 'ditox';
import nodemailer from 'nodemailer';

import { AuthenticationController } from './authentication/authentication.controller';
import { AuthenticationService } from './authentication/authentication.service';
import { RequestAuthenticationLink } from './authentication/commands/request-authentication-link.command';
import { RevokeSessionToken } from './authentication/commands/revoke-session-token.command';
import { VerifyAuthenticationToken } from './authentication/commands/verify-authentication-token.command';
import { SendAuthenticationEmail } from './authentication/event-handlers/send-authentication-email.event-handler';
import { GetAuthenticatedMember } from './authentication/queries/get-authenticated-member.query';
import { GetToken } from './authentication/queries/get-session-token.query';
import { SessionController } from './authentication/session.controller';
import { SessionProvider } from './authentication/session.provider';
import { CommentService } from './comments/comment.service';
import { EnvConfigAdapter } from './infrastructure/config/env-config.adapter';
import { CommandBus } from './infrastructure/cqs/command-bus';
import { EventBus } from './infrastructure/cqs/event-bus';
import { QueryBus } from './infrastructure/cqs/query-bus';
import { NativeDateAdapter } from './infrastructure/date/native-date.adapter';
import { EmailPackageRendererAdapter } from './infrastructure/email/email-package-renderer.adapter';
import { NodemailerEmailSenderAdapter } from './infrastructure/email/nodemailer-email-sender.adapter';
import { SlackErrorReporterAdapter } from './infrastructure/error-reporter/slack-error-reporter.adapter';
import { EventPublisher } from './infrastructure/events/event-publisher';
import { EventsLogger } from './infrastructure/events/events-logger';
import { EventsPersistor } from './infrastructure/events/events-persistor';
import { EventsSlackPublisher } from './infrastructure/events/events-slack-publisher';
import { NanoIdGenerator } from './infrastructure/generator/nanoid-generator.adapter';
import { CheerioHtmlParserAdapter } from './infrastructure/html-parser/cheerio-html-parser.adapter';
import { ConsoleLogger } from './infrastructure/logger/console-logger.adapter';
import { WebPushNotificationAdapter } from './infrastructure/push-notification/web-push-notification.adapter';
import { WebSlackClientAdapter } from './infrastructure/slack/web-slack-client.adapter';
import { FormatJsTranslationAdapter } from './infrastructure/translation/formatjs-translation.adapter';
import { ChangeNotificationDeliveryType } from './members/commands/change-notification-delivery-type.command';
import { CreateMember } from './members/commands/create-member.command';
import { UpdateMemberProfile } from './members/commands/update-member-profile.command';
import { CreateMemberSubscription } from './members/event-handlers/create-member-subscription.event-handler';
import { MembersController } from './members/members.controller';
import { GetMember } from './members/queries/get-member.query';
import { ListMembers } from './members/queries/list-members.query';
import { CreateSubscription } from './notifications/commands/create-subscription.command';
import { MarkNotificationAsRead } from './notifications/commands/mark-notification-as-read.command';
import { Notify } from './notifications/commands/notify.command';
import { RegisterDevice } from './notifications/commands/register-device.command';
import { SendEmailNotification } from './notifications/commands/send-email-notification.command';
import { SendPushNotification } from './notifications/commands/send-push-notification.command';
import { DeliverNotification } from './notifications/event-handlers/deliver-notification.event-handler';
import { NotificationController } from './notifications/notification.controller';
import { GetMemberNotifications } from './notifications/queries/get-member-notifications.query';
import { Database } from './persistence/database';
import { SqlCommentRepository } from './persistence/repositories/comment/sql-comment.repository';
import { SqlMemberRepository } from './persistence/repositories/member/sql-member.repository';
import { SqlMemberDeviceRepository } from './persistence/repositories/member-device/sql-member-device.repository';
import { SqlNotificationRepository } from './persistence/repositories/notification/sql-notification.repository';
import { SqlRequestRepository } from './persistence/repositories/request/sql-request.repository';
import { SqlSubscriptionRepository } from './persistence/repositories/subscription/sql-subscription.repository';
import { SqlTokenRepository } from './persistence/repositories/token/sql-token.repository';
import { CreateRequestComment } from './requests/commands/create-request-comment.command';
import { CreateRequest } from './requests/commands/create-request.command';
import { EditRequest } from './requests/commands/edit-request.command';
import { CreateRequestSubscription } from './requests/event-handlers/create-request-subscriptions.event-handler';
import { NotifyRequestCommentCreated } from './requests/event-handlers/notify-request-comment-created.event-handler';
import { NotifyRequestCreated } from './requests/event-handlers/notify-request-created.event-handler';
import { GetRequest } from './requests/queries/get-request.query';
import { ListRequests } from './requests/queries/list-requests.query';
import { RequestController } from './requests/request.controller';
import { Server } from './server';
import { COMMANDS, EVENT_HANDLERS, QUERIES, TOKENS } from './tokens';

export const container = createContainer();

container.bindValue(TOKENS.container, container);

container.bindFactory(TOKENS.config, EnvConfigAdapter.inject);
container.bindFactory(TOKENS.date, NativeDateAdapter.inject);
container.bindFactory(TOKENS.generator, NanoIdGenerator.inject);
container.bindFactory(TOKENS.logger, ConsoleLogger.inject);
container.bindFactory(TOKENS.slackClient, WebSlackClientAdapter.inject);
container.bindFactory(TOKENS.errorReporter, SlackErrorReporterAdapter.inject);
container.bindFactory(TOKENS.htmlParser, CheerioHtmlParserAdapter.inject);
container.bindFactory(TOKENS.translation, FormatJsTranslationAdapter.inject);
container.bindFactory(TOKENS.pushNotification, WebPushNotificationAdapter.inject);

container.bindFactory(TOKENS.server, Server.inject);
container.bindFactory(TOKENS.database, Database.inject);
container.bindFactory(TOKENS.emailRenderer, EmailPackageRendererAdapter.inject);
container.bindValue(TOKENS.nodemailer, nodemailer);
container.bindFactory(TOKENS.emailSender, NodemailerEmailSenderAdapter.inject);

container.bindFactory(TOKENS.membersController, MembersController.inject);
container.bindFactory(TOKENS.memberRepository, SqlMemberRepository.inject);

container.bindFactory(TOKENS.requestController, RequestController.inject);
container.bindFactory(TOKENS.requestRepository, SqlRequestRepository.inject);

container.bindFactory(TOKENS.authenticationController, AuthenticationController.inject);
container.bindFactory(TOKENS.sessionController, SessionController.inject);
container.bindFactory(TOKENS.sessionProvider, SessionProvider.inject);
container.bindFactory(TOKENS.authenticationService, AuthenticationService.inject);
container.bindFactory(TOKENS.tokenRepository, SqlTokenRepository.inject);

container.bindFactory(TOKENS.commentService, CommentService.inject);
container.bindFactory(TOKENS.commentRepository, SqlCommentRepository.inject);

container.bindFactory(TOKENS.subscriptionRepository, SqlSubscriptionRepository.inject);

container.bindFactory(TOKENS.notificationController, NotificationController.inject);
container.bindFactory(TOKENS.notificationRepository, SqlNotificationRepository.inject);

container.bindFactory(TOKENS.memberDeviceRepository, SqlMemberDeviceRepository.inject);

container.bindFactory(TOKENS.commandBus, CommandBus.inject);
container.bindFactory(TOKENS.queryBus, QueryBus.inject);
container.bindFactory(TOKENS.eventBus, EventBus.inject);
container.bindFactory(TOKENS.eventPublisher, EventPublisher.inject);

container.bindFactory(COMMANDS.requestAuthenticationLink, RequestAuthenticationLink.inject);
container.bindFactory(COMMANDS.verifyAuthenticationToken, VerifyAuthenticationToken.inject);
container.bindFactory(COMMANDS.revokeSessionToken, RevokeSessionToken.inject);
container.bindFactory(COMMANDS.createRequest, CreateRequest.inject);
container.bindFactory(COMMANDS.editRequest, EditRequest.inject);
container.bindFactory(COMMANDS.createRequestComment, CreateRequestComment.inject);
container.bindFactory(COMMANDS.createMember, CreateMember.inject);
container.bindFactory(COMMANDS.updateMemberProfile, UpdateMemberProfile.inject);
container.bindFactory(COMMANDS.changeNotificationDeliveryType, ChangeNotificationDeliveryType.inject);
container.bindFactory(COMMANDS.createSubscription, CreateSubscription.inject);
container.bindFactory(COMMANDS.markNotificationAsRead, MarkNotificationAsRead.inject);
container.bindFactory(COMMANDS.registerDevice, RegisterDevice.inject);
container.bindFactory(COMMANDS.notify, Notify.inject);
container.bindFactory(COMMANDS.sendPushNotification, SendPushNotification.inject);
container.bindFactory(COMMANDS.sendEmailNotification, SendEmailNotification.inject);

container.bindFactory(QUERIES.getToken, GetToken.inject);
container.bindFactory(QUERIES.getAuthenticatedMember, GetAuthenticatedMember.inject);
container.bindFactory(QUERIES.listRequests, ListRequests.inject);
container.bindFactory(QUERIES.getRequest, GetRequest.inject);
container.bindFactory(QUERIES.listMembers, ListMembers.inject);
container.bindFactory(QUERIES.getMember, GetMember.inject);
container.bindFactory(QUERIES.getMemberNotifications, GetMemberNotifications.inject);

container.bindFactory(EVENT_HANDLERS.eventsLogger, EventsLogger.inject);
container.bindFactory(EVENT_HANDLERS.eventsPersistor, EventsPersistor.inject);
container.bindFactory(EVENT_HANDLERS.eventsSlackPublisher, EventsSlackPublisher.inject);
container.bindFactory(EVENT_HANDLERS.sendAuthenticationEmail, SendAuthenticationEmail.inject);
container.bindFactory(EVENT_HANDLERS.createRequestSubscription, CreateRequestSubscription.inject);
container.bindFactory(EVENT_HANDLERS.notifyRequestCreated, NotifyRequestCreated.inject);
container.bindFactory(EVENT_HANDLERS.notifyRequestCommentCreated, NotifyRequestCommentCreated.inject);
container.bindFactory(EVENT_HANDLERS.createMemberSubscription, CreateMemberSubscription.inject);
container.bindFactory(EVENT_HANDLERS.deliverNotification, DeliverNotification.inject);
