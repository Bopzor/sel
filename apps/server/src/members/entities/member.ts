import { createFactory, createId } from '@sel/utils';

import { Address, createAddress } from './address';

export type Member = {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  bio: string;
  address: Address;
};

export const createMember = createFactory<Member>(() => ({
  id: createId(),
  firstName: '',
  lastName: '',
  email: '',
  phoneNumber: '',
  bio: '',
  address: createAddress(),
}));
