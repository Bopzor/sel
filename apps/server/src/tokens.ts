import { Container, token } from 'ditox';

import { AuthenticationController } from './authentication/authentication.controller';
import { AuthenticationFacade } from './authentication/authentication.facade';
import { AuthenticationModule } from './authentication/authentication.module';
import { AuthenticationService } from './authentication/authentication.service';
import { TokenRepository } from './authentication/token.repository';
import { CommentsFacade } from './comments/comments.facade';
import { CommentsRepository } from './comments/comments.repository';
import { CommentsService } from './comments/comments.service';
import { ConfigPort } from './infrastructure/config/config.port';
import { DatePort } from './infrastructure/date/date.port';
import { EmailRendererPort } from './infrastructure/email/email-renderer.port';
import { EmailSenderPort } from './infrastructure/email/email-sender.port';
import { Nodemailer } from './infrastructure/email/nodemailer-email-sender.adapter';
import { ErrorReporterPort } from './infrastructure/error-reporter/error-reporter.port';
import { EventsLogger } from './infrastructure/events/events-logger';
import { EventsPersistor } from './infrastructure/events/events-persistor';
import { EventsSlackPublisher } from './infrastructure/events/events-slack-publisher';
import { EventsPort } from './infrastructure/events/events.port';
import { GeneratorPort } from './infrastructure/generator/generator.port';
import { HtmlParserPort } from './infrastructure/html-parser/html-parser.port';
import { LoggerPort } from './infrastructure/logger/logger.port';
import { Database } from './infrastructure/persistence/database';
import { SlackClientPort } from './infrastructure/slack/slack-client.port';
import { MembersController } from './members/members.controller';
import { MembersFacade } from './members/members.facade';
import { MembersRepository } from './members/members.repository';
import { MembersService } from './members/members.service';
import { Persistor } from './persistor';
import { RequestController } from './requests/request.controller';
import { RequestRepository } from './requests/request.repository';
import { RequestService } from './requests/request.service';
import { Server } from './server';
import { SessionController } from './session/session.controller';
import { SessionProvider } from './session/session.provider';
import { SessionService } from './session/session.service';

export const TOKENS = {
  container: token<Container>('container'),
  config: token<ConfigPort>('config'),
  date: token<DatePort>('date'),
  generator: token<GeneratorPort>('generator'),
  logger: token<LoggerPort>('logger'),
  persistor: token<Persistor>('persistor'),
  events: token<EventsPort>('events'),
  eventsLogger: token<EventsLogger>('eventsLogger'),
  eventsPersistor: token<EventsPersistor>('eventsPersistor'),
  eventsSlackPublisher: token<EventsSlackPublisher>('eventsSlackPublisher'),
  slackClient: token<SlackClientPort>('slackClientPort'),
  errorReporter: token<ErrorReporterPort>('errorReporter'),
  htmlParser: token<HtmlParserPort>('htmlParser'),
  server: token<Server>('server'),
  database: token<Database>('database'),
  nodemailer: token<Nodemailer>('nodemailer'),
  emailRenderer: token<EmailRendererPort>('emailRenderer'),
  emailSender: token<EmailSenderPort>('emailSender'),
  membersFacade: token<MembersFacade>('membersFacade'),
  membersController: token<MembersController>('membersController'),
  membersService: token<MembersService>('membersService'),
  membersRepository: token<MembersRepository>('membersRepository'),
  sessionProvider: token<SessionProvider>('authenticatedMemberProvider'),
  authenticationModule: token<AuthenticationModule>('authenticationModule'),
  authenticationFacade: token<AuthenticationFacade>('authenticationFacade'),
  authenticationController: token<AuthenticationController>('authenticationController'),
  authenticationService: token<AuthenticationService>('authenticationService'),
  tokenRepository: token<TokenRepository>('tokenRepository'),
  sessionController: token<SessionController>('sessionController'),
  sessionService: token<SessionService>('sessionService'),
  requestController: token<RequestController>('requestController'),
  requestService: token<RequestService>('requestService'),
  requestRepository: token<RequestRepository>('requestRepository'),
  commentsFacade: token<CommentsFacade>('commentsFacade'),
  commentsService: token<CommentsService>('commentsService'),
  commentsRepository: token<CommentsRepository>('commentsRepository'),
};
