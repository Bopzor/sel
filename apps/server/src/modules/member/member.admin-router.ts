import * as shared from '@sel/shared';
import express from 'express';

import { db } from 'src/persistence';

import { File } from '../file/file.entity';

import { Member } from './member.entities';

export const router = express.Router();

router.get('/', async (req, res) => {
  const members = await db.query.members.findMany({
    with: {
      avatar: true,
    },
  });

  res.json(members.map(serializeAdminMember));
});

function serializeAdminMember(member: Member & { avatar: File | null }): shared.AdminMember {
  return {
    id: member.id,
    status: member.status,
    firstName: member.firstName,
    lastName: member.lastName,
    number: member.number,
    email: member.email,
    phoneNumbers: member.phoneNumbers,
    avatar: member.avatar?.name,
    balance: member.balance,
  };
}
