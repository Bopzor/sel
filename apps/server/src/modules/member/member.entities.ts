import { DomainEvent } from 'src/infrastructure/events';
import { type schema } from 'src/persistence';

import { NotificationDeliveryType } from '../notification/notification.entities';

export type Member = typeof schema.members.$inferSelect;
export type MemberInsert = typeof schema.members.$inferInsert;

export class MemberCreatedEvent extends DomainEvent {}

export class OnboardingCompletedEvent extends DomainEvent {}

export class NotificationDeliveryTypeChangedEvent extends DomainEvent<{
  notificationDeliveryType: Partial<Record<NotificationDeliveryType, boolean>>;
}> {}
