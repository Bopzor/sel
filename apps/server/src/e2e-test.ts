import { defined } from '@sel/utils';
import { expect } from 'vitest';

import { AuthenticationService } from './authentication/authentication.service';
import { Token, TokenType } from './authentication/token.entity';
import { container } from './container';
import { StubConfigAdapter } from './infrastructure/config/stub-config.adapter';
import { TestMailSever } from './infrastructure/email/test-mail-server';
import { TestErrorReporterAdapter } from './infrastructure/error-reporter/test-error-reporter.adapter';
import { StubLogger } from './infrastructure/logger/stub-logger.adapter';
import { Member } from './members/entities';
import { MembersRepository } from './members/members.repository';
import { MembersService } from './members/members.service';
import { Request } from './requests/request.entity';
import { RequestRepository } from './requests/request.repository';
import { RequestService } from './requests/request.service';
import { TOKENS } from './tokens';

export class E2ETest {
  config = new StubConfigAdapter({
    server: {
      host: 'localhost',
      port: 3030,
    },
    session: {
      secret: '',
      secure: false,
    },
    database: {
      url: process.env.DATABASE_URL ?? 'postgres://postgres@localhost/test',
    },
    app: {
      baseUrl: 'http://app.url',
    },
    email: {
      host: 'localhost',
      password: '',
      secure: false,
      port: 1125,
      sender: 'sel@localhost',
      templatesPath: 'email-templates',
    },
  });

  logger = new StubLogger();
  errorReporter = new TestErrorReporterAdapter();

  mailServer = new TestMailSever(this.config);

  private get server() {
    return container.resolve(TOKENS.server);
  }

  create!: EntityCreator;

  async setup() {
    container.bindValue(TOKENS.config, this.config);
    container.bindValue(TOKENS.logger, this.logger);
    container.bindValue(TOKENS.errorReporter, this.errorReporter);

    await container.resolve(TOKENS.database).reset();
    await container.resolve(TOKENS.emailRenderer).init?.();
    container.resolve(TOKENS.authenticationModule).init();
    container.resolve(TOKENS.membersModule).init();
    container.resolve(TOKENS.requestModule).init();

    this.create = new EntityCreator(
      container.resolve(TOKENS.membersService),
      container.resolve(TOKENS.membersRepository),
      container.resolve(TOKENS.authenticationService),
      container.resolve(TOKENS.requestService),
      container.resolve(TOKENS.requestRepository)
    );

    await this.mailServer.listen();
    await this.server.start();
  }

  async teardown() {
    await this.server.close();
    await this.mailServer.close();
  }

  async fetch(
    path: string,
    options: { method?: string; token?: string; assertStatus?: boolean; body?: unknown } = {}
  ) {
    const { method = 'GET', token, assertStatus = true, body: requestBody } = options;

    let response: Response | undefined = undefined;
    let body: unknown = undefined;

    try {
      const headers = new Headers();
      const init: RequestInit = { method, headers };

      if (token) {
        headers.set('Cookie', `token=${token}`);
      }

      if (requestBody) {
        init.body = JSON.stringify(requestBody);
        headers.set('Content-Type', 'application/json');
      }

      response = await fetch(`http://localhost:3030${path}`, init);

      if (response.headers.get('Content-Type')?.startsWith('application/json')) {
        body = await response.clone().json();
      } else {
        body = await response.clone().text();
      }

      if (assertStatus) {
        expect(response.ok).toBe(true);
      }

      return { response, body };
    } catch (error) {
      if (body) {
        // eslint-disable-next-line no-console
        console.log(body);
      }

      throw error;
    }
  }
}

class EntityCreator {
  constructor(
    private readonly memberService: MembersService,
    private readonly memberRepository: MembersRepository,
    private readonly authenticationService: AuthenticationService,
    private readonly requestService: RequestService,
    private readonly requestRepository: RequestRepository
  ) {}

  // prettier-ignore
  async member({ firstName = '', lastName = '', email = '' }: { firstName?: string; lastName?: string; email?: string } = {}): Promise<Member> {
    const memberId = await this.memberService.createMember(firstName, lastName, email);
    return defined(await this.memberRepository.getMember(memberId));
  }

  async token(type: TokenType, memberId: string): Promise<Token> {
    return this.authenticationService.generateToken(type, memberId);
  }

  // prettier-ignore
  async request({requesterId = '', title = '', body = '' }: { requesterId?: string, title?: string, body?: string } = {}): Promise<Request> {
    const requestId = await this.requestService.createRequest(requesterId, title, body);
    return defined(await this.requestRepository.getRequest(requestId));
  }
}
