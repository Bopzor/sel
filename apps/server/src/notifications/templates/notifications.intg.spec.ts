import * as shared from '@sel/shared';
import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, it } from 'vitest';

import { NotificationDeliveryType } from '../../common/notification-delivery-type';
import { container } from '../../container';
import { E2ETest } from '../../e2e-test';
import { Member } from '../../members/member.entity';
import { TOKENS } from '../../tokens';

class Test extends E2ETest {
  member!: Member;

  get notificationService() {
    return container.resolve(TOKENS.notificationService);
  }

  async setup(): Promise<void> {
    this.member = await this.create.member({ email: 'member@email', firstName: 'Member' });

    await this.application.changeNotificationDeliveryType({
      memberId: this.member.id,
      notificationDeliveryType: {
        [NotificationDeliveryType.push]: true,
        [NotificationDeliveryType.email]: true,
      },
    });

    await this.registerDevice(this.member, 'subscription');
  }

  async registerDevice(member: Member, subscription: string) {
    await this.application.registerDevice({
      memberId: member.id,
      deviceSubscription: subscription,
      deviceType: 'mobile',
    });
  }

  async assertNotificationSnapshot<Type extends shared.NotificationType>(
    type: Type,
    context: shared.NotificationData[Type],
  ) {
    await this.notificationService.notify([this.member.id], type, () => context);

    expect(this.logger.lines.error, this.logger.lines.error.join('\n')).toHaveLength(0);
    expect(this.pushNotification.notifications.get('"subscription"')).toMatchSnapshot();
    expect(this.emails).toMatchSnapshot();
  }

  private get emails() {
    return this.mailServer.emails.map((email) => ({
      to: email.to[0].address,
      subject: email.subject,
      text: email.text,
      html: email.html
        ?.replaceAll('\n', '')
        .replaceAll(/.*<!-- CONTENT START -->|<!-- CONTENT END -->.*/g, ''),
    }));
  }
}

describe('[intg] Notification snapshots', () => {
  let test: Test;

  beforeAll(async () => {
    test = await Test.create(Test);
  });

  afterAll(() => test?.teardown());

  beforeEach(() => test.reset());
  afterEach(() => test?.waitForEventHandlers());

  it('Test', async () => {
    await test.assertNotificationSnapshot('Test', {
      member: { firstName: 'Member' },
      answer: 42,
      content: {
        html: '<p>hello <strong>world</strong></p>',
        text: 'hello world',
      },
    });
  });

  it('RequestCreated', async () => {
    await test.assertNotificationSnapshot('RequestCreated', {
      member: { firstName: 'Member' },
      request: {
        id: 'requestId',
        title: 'Request title',
        requester: {
          id: 'requesterId',
          name: 'Requester',
        },
        body: {
          html: '<p>Request body</p>',
          text: 'Request body',
        },
      },
    });
  });

  it('RequestCommentCreated', async () => {
    await test.assertNotificationSnapshot('RequestCommentCreated', {
      member: { firstName: 'Member' },
      isRequester: true,
      request: {
        id: 'requestId',
        title: 'Request title',
        requester: {
          id: 'requesterId',
          name: 'Requester',
        },
      },
      comment: {
        id: 'commentId',
        author: {
          id: 'authorId',
          name: 'Author',
        },
        body: {
          html: '<p>Comment body</p>',
          text: 'Comment body',
        },
      },
    });
  });

  it('RequestStatusChanged', async () => {
    await test.assertNotificationSnapshot('RequestStatusChanged', {
      member: { firstName: 'Member' },
      request: {
        id: 'requestId',
        title: 'Request title',
        status: shared.RequestStatus.canceled,
        requester: {
          id: 'requesterId',
          name: 'Requester',
        },
      },
    });
  });
});
