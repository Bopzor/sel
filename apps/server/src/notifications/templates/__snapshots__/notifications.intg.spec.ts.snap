// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[intg] Notification snapshots > EventCommentCreated 1`] = `
[
  {
    "content": "Author : Comment body",
    "link": "http://app.url/events/eventId",
    "title": "Nouveau commentaire sur votre √©v√©nement "Event title"",
  },
]
`;

exports[`[intg] Notification snapshots > EventCommentCreated 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Author</strong> a √©crit un commentaire sur votre √©v√©nement <a href="http://app.url/events/eventId">Event title</a> :</p><blockquote style="margin: 0; padding: 8px 16px; border: 1px solid #CCC; border-radius: 4px; background: #fafafc;"><p>Comment body</p></blockquote>",
    "subject": "Commentaire de Author sur votre √©v√©nement "Event title"",
    "text": "Bonjour Member,

Author a √©crit un commentaire sur votre √©v√©nement Event title (http://app.url/events/eventId) :

Comment body",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > EventCreated 1`] = `
[
  {
    "content": "Event title",
    "link": "http://app.url/events/eventId",
    "title": "Nouvel √©v√©nement",
  },
]
`;

exports[`[intg] Notification snapshots > EventCreated 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Organizer</strong> a cr√©√© un nouvel √©v√©nement : <a href="http://app.url/events/eventId">Event title</a></p><blockquote style="margin: 0; padding: 8px 16px; border: 1px solid #CCC; border-radius: 4px; background: #fafafc;"><p>Event body</p></blockquote>",
    "subject": "Nouvel √©v√©nement : Event title",
    "text": "Bonjour Member,

Organizer a cr√©√© un nouvel √©v√©nement : Event title (http://app.url/events/eventId)

Event body",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > EventParticipationSet 1`] = `
[
  {
    "content": "Event title",
    "link": "http://app.url/events/eventId",
    "title": "Participation √† votre √©v√©nement Event title",
  },
]
`;

exports[`[intg] Notification snapshots > EventParticipationSet 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Participant</strong> participera √† votre √©v√©nement <a href="http://app.url/events/eventId">Event title</a>.</p>",
    "subject": "Participation √† votre √©v√©nement Event title",
    "text": "Bonjour Member,

Participant participera √† votre √©v√©nement Event title (http://app.url/events/eventId).",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > RequestCommentCreated 1`] = `
[
  {
    "content": "Author : Comment body",
    "link": "http://app.url/requests/requestId",
    "title": "Nouveau commentaire sur votre demande "Request title"",
  },
]
`;

exports[`[intg] Notification snapshots > RequestCommentCreated 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Author</strong> a √©crit un commentaire sur votre demande <a href="http://app.url/requests/requestId">Request title</a> :</p><blockquote style="margin: 0; padding: 8px 16px; border: 1px solid #CCC; border-radius: 4px; background: #fafafc;"><p>Comment body</p></blockquote>",
    "subject": "Commentaire de Author sur votre demande "Request title"",
    "text": "Bonjour Member,

Author a √©crit un commentaire sur votre demande Request title (http://app.url/requests/requestId) :

Comment body",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > RequestCreated 1`] = `
[
  {
    "content": "Request title",
    "link": "http://app.url/requests/requestId",
    "title": "Demande de Requester",
  },
]
`;

exports[`[intg] Notification snapshots > RequestCreated 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Requester</strong> a publi√© une nouvelle demande : <a href="http://app.url/requests/requestId">Request title</a></p><blockquote style="margin: 0; padding: 8px 16px; border: 1px solid #CCC; border-radius: 4px; background: #fafafc;"><p>Request body</p></blockquote>",
    "subject": "Demande de Requester : Request title",
    "text": "Bonjour Member,

Requester a publi√© une nouvelle demande : Request title (http://app.url/requests/requestId)

Request body",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > RequestStatusChanged 1`] = `
[
  {
    "content": "Request title",
    "link": "http://app.url/requests/requestId",
    "title": "Requester a annul√© sa demande",
  },
]
`;

exports[`[intg] Notification snapshots > RequestStatusChanged 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Requester</strong> a annul√© sa demande <a href="http://app.url/requests/requestId">Request title</a></p>",
    "subject": "Requester a annul√© sa demande "Request title"",
    "text": "Bonjour Member,

Requester a annul√© sa demande Request title (http://app.url/requests/requestId)",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > Test 1`] = `
[
  {
    "content": "Hello Member, this is a test notification! I hope it works üòÑ",
    "link": "http://app.url",
    "title": "Test notification (push)",
  },
]
`;

exports[`[intg] Notification snapshots > Test 2`] = `
[
  {
    "html": "<p>Hello Member, this is <strong>a test notification</strong>! I hope it works üòÑ</p><p>Here is <a href="http://app.url">a link</a>.</p><p>42 is a good answer.</p><p>And here is some verbatim content:</p><blockquote style="margin: 0; padding: 8px 16px; border: 1px solid #CCC; border-radius: 4px; background: #fafafc;"><p>hello <strong>world</strong></p></blockquote><p>Cheers!</p>",
    "subject": "Test notification (email)",
    "text": "Hello Member, this is a test notification! I hope it works üòÑ

Here is a link (http://app.url).

42 is a good answer.

And here is some verbatim content:

hello world

Cheers!",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > TransactionCanceled 1`] = `
[
  {
    "content": "Payer n'a pas accept√© votre demande de coins pour l'√©change : Transaction description",
    "link": "http://app.url/profile/transactions?transactionId=transactionId",
    "title": "√âchange avec Payer",
  },
]
`;

exports[`[intg] Notification snapshots > TransactionCanceled 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Payer</strong> a annul√© l'√©change : Transaction description.</p>",
    "subject": "Payer n'a pas accept√© votre demande de coins pour l'√©change : Transaction description",
    "text": "Bonjour Member,

Payer a annul√© l'√©change : Transaction description.",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > TransactionCompleted 1`] = `
[
  {
    "content": "Vous avez re√ßu 123 coins pour l'√©change : Transaction description",
    "link": "http://app.url/profile/transactions?transactionId=transactionId",
    "title": "√âchange avec Payer",
  },
]
`;

exports[`[intg] Notification snapshots > TransactionCompleted 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Payer</strong> a accept√© votre demande de <strong>123 coins</strong> pour l'√©change : <a href="http://app.url/profile/transactions?transactionId=transactionId">Transaction description</a>.</p>",
    "subject": "Vous avez re√ßu 123 coins pour l'√©change : Transaction description",
    "text": "Bonjour Member,

Payer a accept√© votre demande de 123 coins pour l'√©change : Transaction description (http://app.url/profile/transactions?transactionId=transactionId).",
    "to": "member@email",
  },
]
`;

exports[`[intg] Notification snapshots > TransactionPending 1`] = `
[
  {
    "content": "Recipient vous demande 123 coins pour l'√©change : Transaction description",
    "link": "http://app.url/profile/transactions?transactionId=transactionId",
    "title": "√âchange avec Recipient",
  },
]
`;

exports[`[intg] Notification snapshots > TransactionPending 2`] = `
[
  {
    "html": "<p>Bonjour Member,</p><p><strong>Recipient</strong> vous demande <strong>123 coins</strong> pour l'√©change : Transaction description.</p><p>Vous pouvez valider ou refuser cet √©change depuis <a href="http://app.url/profile/transactions?transactionId=transactionId">votre page de profil</a>.</p>",
    "subject": "Demande de coins pour l'√©change : Transaction description",
    "text": "Bonjour Member,

Recipient vous demande 123 coins pour l'√©change : Transaction description.

Vous pouvez valider ou refuser cet √©change depuis votre page de profil (http://app.url/profile/transactions?transactionId=transactionId).",
    "to": "member@email",
  },
]
`;
